src/runtime/hexagon/hexagon_buffer_manager.h

```diff
@@ -85,6 +86,18 @@ class HexagonBufferManager {
     return hexagon_buffer_map_.empty();
   }
 
+  //! \brief Returns a list of allocated pointers.
+  // Note - this should only be used by the device API to keep track of what
+  // was in the manager when HexagonDeviceAPI::ReleaseResources is called.
+  std::vector<void*> vector() {
+    std::vector<void*> allocated;
+    std::lock_guard<std::mutex> lock(map_mutex_);
+    for (auto it = hexagon_buffer_map_.begin(); it != hexagon_buffer_map_.end(); it++) {
```

Comment:
FYI, since we're on C++17 now, structured bindings with a ranged-based for loop reduce a lot of the iterator-based boilerplate.  This is more personal style, but it would let the loop be re-written as below.

```c++
for(const auto& [data_ptr, buffer] : hexagon_buffer_map_) {
    allocated.push_back(data_ptr);
}
```

Reply:
I like that much better, thanks!